src/Core/Form/FormChoiceFormatter.php

```diff
@@ -0,0 +1,110 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Form;
+
+/**
+ * Class that formats choices for Symfony forms.
+ */
+class FormChoiceFormatter
+{
+    /**
+     * Returns a choice list in a format required for Symfony form.
+     * Symfony form choice fields accept array with options, where the item name
+     * is the key and item ID is the value.
+     *
+     * So, when there are two items with the same name, they get lost.
+     * This will automatically mark duplicate options with their ID.
+     *
+     * @param array $rawChoices Raw array with data to build the options from
+     * @param string $idKey Key name of the item IDs, id_carrier for example
+     * @param string $nameKey Key name of the item NAMEs, carrier_name for example
+     * @param bool $sortByName Should the list be automatically sorted by name
+     *
+     * @return array Formatted choices
+     */
+    public static function formatFormChoices(array $rawChoices, string $idKey, string $nameKey, bool $sortByName = true): array
```

Comment:
I think the method name is a bit redundant with the class name. `FormChoiceFormatter::format` would be sufficient